File: Server Group Updates/GroupUpdates.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Web;
using Microsoft.AspNet.SignalR;

namespace Server_Group_Updates
{
    public class GroupUpdates : Hub
    {
        //Variables
        private static List<string> groupNames = new List<string>();
        private static int currentConnections = 0;
        private string groupName;

        public void SendGroupMessage(string pFrom, string pMessage, string pGroupName)
        {
            Clients.Group(pGroupName).broadcastMessage(pFrom, pGroupName, pMessage);
        }

        public void UpdateGroupName(string pGroupName)
        {
            Groups.Add(Context.ConnectionId, pGroupName);
            groupNames.Add(pGroupName);
            groupName = pGroupName;

            Clients.All.deleteAllGroups();

            SendAllGroups();

            Clients.All.broadcastMessage("ADMIN", "ALL", pGroupName + " has joined.");
        }

        public void AddGroupToList(string pGroupName)
        {
            Clients.All.addGroupToList(pGroupName);
        }

        public void SendAllGroups()
        {
            for (int i = 1; i <= groupNames.Count(); i++)
            {
                AddGroupToList(groupNames[i-1]);
            }
        }

        public void SendGroupsData()
        {
            for (int i = 0; i <= 9; i++)
            {
                for (int x = 1; x <= groupNames.Count(); x++)
                {
                    Clients.Group(groupNames[x-1]).broadcastMessage("ADMIN", groupNames[x-1], "Message " + i.ToString() + " from Admin");
                }
            }
        }

        public void ReturnValue(string pValue)
        {
            Clients.Caller.returnValue(pValue);
        }

        public override Task OnConnected()
        {
            currentConnections++;

            Clients.All.updateConnectionCount(currentConnections);

            return base.OnConnected();
        }

        public override Task OnDisconnected()
        {
            if (groupName != null)
            {
                Groups.Remove(Context.ConnectionId, groupName);
                currentConnections--;
            }

            Clients.All.updateConnectionCount(currentConnections);

            return base.OnDisconnected();
        }
    }
}

File: Server Group Updates/js/GroupUpdates.js
$(document).ready(function () {
    //Variables
    var proxy = $.connection.groupUpdates;
    var connected = false;
    var groupName;

    //Logging
    //$.connection.hub.logging = true;

    //Server side functions

    //Client side functions
    proxy.client.updateConnectionCount = function (x) {
        $("#connections").text(x);
    }

    proxy.client.broadcastMessage = function (from, grp, msg) {
        $("ul").append("<li><b>" + from + "</b> -> <b>" + grp + "</b> : " + msg + "</li>");
    }

    proxy.client.addGroupToList = function (grp) {
        if ($("select").index("<option value='" + grp + "'>" + grp + "</option>") == -1) {
            $("select").append("<option value='" + grp + "'>" + grp + "</option>");
        }
    }

    proxy.client.deleteAllGroups = function () {
        $("#groups option").each(function () {
            $(this).remove();
        });
    }

    //Event handlers
    $("button#send-message").click(function () {
        var group;
        var message;

        group = $("#groups").val();
        message = $("input#message-content").val();

        proxy.server.sendGroupMessage(groupName, message, group);

        $("input#message-content").val("");
        $("input#message-content").focus();
    });

    $("button#send-data").click(function () {
        proxy.server.sendGroupsData();
    });

    // Connection
    $.connection.hub.start()
        .done(function () {
            groupName = prompt("Choose a group name", "");

            if (groupName != null && groupName != "") {
                proxy.server.updateGroupName(groupName);
                $("#yourGroupName").text(groupName);
                connected = true;
            }
            else {
                var randName = "AnonGroup" + Math.floor(Math.random() * 1000000).toString();

                proxy.server.updateGroupName(randName);
                $("#yourGroupName").text(randName);
                connected = true;
            }
    });
});